### ISSUE TITLE
Unable to get a featured image URL

### ISSUE THREAD

**Issue by brnteka** [bueltge/multisite-global-media#25]
After adding a featured image to a post from global media library, i am able to get it with -- the_post_thumbnail() -- from main query, but i'm having troubles getting URL of it and -- get_the_post_thumbnail_url($post_id, 'full') or wp_get_attachment_image_src( get_post_thumbnail_id( $post->ID ), 'large' ); -- returns nothing. Am i missing something?

**Comment by bueltge** [comment]
Please test the new release https://github.com/bueltge/multisite-global-media/releases/tag/0.0.6 and get feedback. Thx.

**Comment by brnteka** [comment]
Great, featured images now behave as expected, but as i found out ACF(Advanced Custom Fields plugin) fields are not saving images choosen from Global Media. ACF field type is set to "Image". Do you think the issue lays on ACF side?

**Comment by bueltge** [comment]
@brnteka not sure, but I think is an problem more globally. Maybe you can debug or show inside the console of the browser to identify errors.

**Comment by mrhasbean** [comment]
Thanks for the update!

Unfortunately, the images still don't work with page builders (specifically, in my case, Beaver Builder and its add-ons - UABB & PowerPack). When I first tried MGM and it didn't work with the Builder, I reported it to the guys at IdeaBox Creations, who produce one of those add-on packs, and this was their response:

"It looks like the "Feature Image" function that we use from WP core isn't returning the correct URL and hence the fallback image is used."

They also asked:

"To rule out the issue, can you try to use BB's Post Module and see if the feature image shows up? 
If yes then please send me WP admin and FTP login details so that I can check what we can do to get it working for PowerPack modules."

Which I did, and it works with a page that isn't using the page builder. So I suspect their suggestion is probably why it isn't working with ACF (or Pods - I tested Pods, doesn't work) too.

**Comment by brnteka** [comment]
WP debug gives this error while trying to save image from Global Media to ACF image field. 

`PHP Notice:  Trying to get property of non-object in /wp-admin/includes/post.php on lines 1390, 1392, 1397, 1431, 1448`

Browsers console is clear. I'm running latest WP 4.9.8 version.

**Comment by mrhasbean** [comment]
As an aside, I have considered forking this excellent project and changing the functionality slightly. I really like the elegance of how this works, it just doesn't quite fit my requirements. For my installations, instead of referencing the image in the "library" blog, it would be more useful to bring a copy of the image to the current blog. I have a couple of reasons for this:

1. For client sites, I don't want them to be able to edit the original image
2. I have custom image sizes for some CPT's, which aren't created on the "library" blog when the image is uploaded, and therefore aren't available
3. Having a copy of the image in the current blog will allow it to work with ACF / Pods, page builders, etc

I envisioned it working something like this:

![add-media-layout](https://user-images.githubusercontent.com/35642539/44345827-bbc56580-a4d7-11e8-8575-7759df710e64.jpg)
![featured-image-layout](https://user-images.githubusercontent.com/35642539/44345830-bbc56580-a4d7-11e8-9b09-e74b7bd1e0b4.jpg)

When the Global Media library is selected, the Insert / Set Featured Image button is disabled, and instead of the image details in the box sidebar, a single button to activate the copy. Once copied, the images would be treated like they've just been uploaded by the user.

Sorry for going a little off-topic. :)

**Comment by bueltge** [comment]
@mrhasbean Maybe you will add your code to copy an image to a site of the network as additional feature, maybe with a filter to un-active this feature? I think it is also helpful for others, especially there use 3th party plugins that will not work with this idea of one media library.

**Comment by bueltge** [comment]
We have a new version, maybe you can test again? @brnteka @mrhasbean 
https://github.com/bueltge/multisite-global-media/releases/tag/0.0.7

**Comment by bueltge** [comment]
New change with #46 and maybe new feedback from your side?

**Comment by bueltge** [comment]
The new version is [0.0.8](https://github.com/bueltge/multisite-global-media/releases/tag/0.0.8) and needs your tests and your feedback here.

**Comment by EdithAllison** [comment]
This may be the same issue: 

I've tested the plugin on a multistore with WooCommerce and the featured product image is not appearing for me in front (tested with Storefront & Twentyseven themes), but it does show in the admin as featured image. 

It is however working fine with posts, so the error appears specific to the WooCommerce plugin. 

The code for the product image is:

/woocommerce/templates/single-product/product-image.php
```
if ( has_post_thumbnail() ) {
  $html  = wc_get_gallery_image_html( $post_thumbnail_id, true );
} 
```

/woocommerce/includes/wc-template-functions.php 
```
function wc_get_gallery_image_html( $attachment_id, $main_image = false ) {
	$flexslider        = (bool) apply_filters( 'woocommerce_single_product_flexslider_enabled', get_theme_support( 'wc-product-gallery-slider' ) );
	$gallery_thumbnail = wc_get_image_size( 'gallery_thumbnail' );
	$thumbnail_size    = apply_filters( 'woocommerce_gallery_thumbnail_size', array( $gallery_thumbnail['width'], $gallery_thumbnail['height'] ) );
	$image_size        = apply_filters( 'woocommerce_gallery_image_size', $flexslider || $main_image ? 'woocommerce_single' : $thumbnail_size );
	$full_size         = apply_filters( 'woocommerce_gallery_full_size', apply_filters( 'woocommerce_product_thumbnails_large_size', 'full' ) );
	$thumbnail_src     = wp_get_attachment_image_src( $attachment_id, $thumbnail_size );
	$full_src          = wp_get_attachment_image_src( $attachment_id, $full_size );
	$image             = wp_get_attachment_image( $attachment_id, $image_size, false, array(
		'title'                   => get_post_field( 'post_title', $attachment_id ),
		'data-caption'            => get_post_field( 'post_excerpt', $attachment_id ),
		'data-src'                => $full_src[0],
		'data-large_image'        => $full_src[0],
		'data-large_image_width'  => $full_src[1],
		'data-large_image_height' => $full_src[2],
		'class'                   => $main_image ? 'wp-post-image' : '',
	) );

	return '<div data-thumb="' . esc_url( $thumbnail_src[0] ) . '" class="woocommerce-product-gallery__image"><a href="' . esc_url( $full_src[0] ) . '">' . $image . '</a></div>';
}
```
Output is

```
<div data-thumb="" class="woocommerce-product-gallery__image">
<a href=""></a>
</div>	
```

Seems related, issue with wp_get_attachment_image_src() - does this mean the plugin is incompatible with WooCommerce?

**Comment by bueltge** [comment]
The woo problem is discussed in #27. So I understand it works with the current version with post, page for you, but not for the products from Woo. @widoz works on a Woo enhancement, so that it should work in a feature version.

**Comment by EdithAllison** [comment]
Correct, posts & pages are fine, it's the WooCommerce product featured image that's not behaving. Will keep following, if Woo can work, then maybe I can get it to work with WP All Import to allow CSV imports. Currently I'm using a workaround with "remote URL" for the sub sites to avoid duplication.

**Comment by widoz** [comment]
@EdithAllison Do you still experience on issues about WC?

**Comment by bueltge** [comment]
I close this issue, no feedback on the discussions and tests. If the problem is still existent, please feel free to reopne and leave you actual feedback, test results. Thx.

**Comment by EdithAllison** [comment]
Apologies for very late reply. Just to confirm that we're now using Multisite Global Media on a live multisite with WooCommerce and works very well. Issue with Woo is resolved as per [#27](https://github.com/bueltge/multisite-global-media/issues/27) . Both featured image & gallery images are working.
